### PR TITLE
[DO NOT MERGE - PROTOTYPE] [Issue #22] Prototype download endpoint

### LINKED ISSUE #22: [Task]: (Backend) Prototype a download endpoint
### Summary

The current experience returns either a zip file or individual file directly from the database.

Make a quick hacky prototype to verify how this might work. Actual implementation would be done as a follow-up

### Acceptance criteria

_No response_

### PR DESCRIPTION
**Pull request by chouinar**
PROTOTYPE for discussion - will not be merged

## Summary
Fixes #22

## Changes proposed
Setup an endpoint that returns a presigned S3 URL (as a redirect)

Localstack setup to have a mocked S3

## Context for reviewers
Experimenting with a few ideas about how we handle a download endpoint. The simplest way I can think of is to put the files on S3, and return a presigned URL. Presigned URLs let users directly interact with S3 files, so we aren't actually handling the transfer of files, the users could download it directly from the source in a safe/secure manner.

The exact way we return the URL we can sort out later, while I did set this up as a redirect, I don't think that actually would be what we'd want. I think instead returning the URL in an ordinary response would be better. The frontend (or any users of the endpoint) could then download it in whatever manner they want. 

## Additional information
This is clunky to run, requires a bit of manual work, if you'd like to test this out, let me know and I'll help.

### CODE REVIEW

File: api/docker-compose.yml [hunk 1]
```diff
@@ -47,6 +47,22 @@ services:
...
+      - "127.0.0.1:4510-4559:4510-4559"  # external services port range
+    environment:
+      # LocalStack configuration: https://docs.localstack.cloud/references/configuration/
+      - DEBUG=${DEBUG:-0}
+      # To improve startup time, only add services we use
+      - SERVICES=s3
+      - EAGER_SERVICES_LOADING=1
+    volumes:
+      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+      - "/var/run/docker.sock:/var/run/docker.sock"
```
Comment (rylew1) [inline api/docker-compose.yml:64]: Nice! Maybe make sure to document specific LocalStack config or requirements in the project? Though it looks all baked into the image here

We won't need a license then?

File: api/Makefile [hunk 2]
```diff
@@ -129,6 +129,10 @@ start-db:
 ## Destroy current DB, setup new one
 db-recreate: clean-volumes init-db
 
+# TODO
+start-localstack:
+	docker-compose up --detach localstack
+
```
Comment (rylew1) [inline api/Makefile:135]: Maybe add a brief comment on the command here on what we use localstack for

File: api/src/api/opportunities_v1/opportunity_routes.py [hunk 3]
```diff
@@ -63,4 +63,36 @@ def opportunity_get(db_session: db.Session, opportunity_id: int) -> response.Api
...
+
+    url = s3.generate_presigned_url("get_object", Params={"Bucket": "myfunbucket", "Key": "myfile.txt"})
+
+    print(url)
+    # TODO - because we're hitting localstack inside the docker container, the URL generated is for inside
+    # to make that work, we need to fix the path - I'm sure there is a better way to do this (changing the endpoint URL to the outside one?)
+    x = url.replace("localstack", "localhost")
+    print(x)
+
+    return redirect(x, 302)
```
Comment (rylew1) [inline api/src/api/opportunities_v1/opportunity_routes.py:98]: Will you write tests covering this endpoint? Also any logging we should do here? or exceptions that may occur on this route?

File: api/Makefile (hunk 2 above)
Comment (rylew1) [reply api/Makefile:135]: Also something in the docs that explains how to spin this up and test grabbing a file locally?

Comment (rylew1) [reply api/Makefile:135]: Also should this command automatically get called on `make start` when you're in local?

Comment (chouinar) [reply api/Makefile:135]: This was just a prototype - would add docs/connect properly in the makefile if I was going to merge this.

File: api/docker-compose.yml (hunk 1 above)
Comment (chouinar) [reply api/docker-compose.yml:64]: Localstack has a community version that is free. Seems the main difference is which features are supported. S3 seems like its fully supported in community, so would work.

**Comment by chouinar** [comment]
Made an actual implementation here: https://github.com/navapbc/simpler-grants-gov/pull/87 - so don't need this prototype anymore

**Pull request closed without merging**